packages/astro/src/@types/astro.ts

```diff
@@ -1555,10 +1555,19 @@ export interface AstroUserConfig {
  */
 export type InjectedScriptStage = 'before-hydration' | 'head-inline' | 'page' | 'page-ssr';
 
+/**
+ * IDs for different priorities of injected routes and redirects:
+ * - "above-project": Override any project route in case of conflict.
+ * - "same-as-project": Behave the same as if the route was defined in the project, following the same priority rules.
+ * - "below-project": Only match if no project route matches.
+ */
+export type RoutePriorityOverride = 'above-project' | 'same-as-project' | 'below-project';
```

Comment:
If this is only applied to injected routes, we should choose a name that tells us that, e.g. `InjectedRoutePriorityOverride`